src/test/java/io/confluent/connect/jdbc/integration/sink/JdbcSinkConnectorIT.java

```diff
@@ -0,0 +1,214 @@
+    startPumbaPauseContainer();
+    sendTestDataToKafka(0, numRecords);
+    ConsumerRecords<byte[], byte[]> totalRecords = connect.kafka().consume(
+        numRecords,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    log.info("Number of records added in kafka {}", totalRecords.count());
+
+    // Add mysql dialect related configurations.
+    props = getConnectorConfigurations();
+
+    // Configure Connector and wait some specific time to start the connector.
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForConnectorToStart(CONNECTOR_NAME, Integer.valueOf(MAX_TASKS));
+
+    // Wait Connector to write data into Mysql
+    waitForConnectorToWriteDataIntoMysql(
+        connection,
+        CONNECTOR_NAME,
+        Integer.valueOf(MAX_TASKS),
+        KAFKA_TOPIC,
+        numRecords);
+    assertRecordsCountAndContent(numRecords);
+    pumbaPauseContainer.close();
+  }
+
+  @Test
+  public void testForDbServerDelay() throws Exception {
+    int numRecords = 500000;
+    // Starting 'pumba' container to periodically delay services in sql container.
+    startPumbaDelayContainer();
+    sendTestDataToKafka(0, numRecords);
+    ConsumerRecords<byte[], byte[]> totalRecords = connect.kafka().consume(
+        numRecords,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    log.info("Number of records added in kafka {}", totalRecords.count());
+
+    // Add mysql dialect related configurations.
+    props = getConnectorConfigurations();
+
+    // Configure Connector and wait some specific time to start the connector.
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForConnectorToStart(CONNECTOR_NAME, Integer.valueOf(MAX_TASKS));
+
+    // Wait Connector to write data into Mysql
+    waitForConnectorToWriteDataIntoMysql(
+        connection,
+        CONNECTOR_NAME,
+        Integer.valueOf(MAX_TASKS),
+        KAFKA_TOPIC,
+        numRecords);
+    assertRecordsCountAndContent(numRecords);
+    pumbaDelayContainer.close();
+  }
+
+  private void assertRecordsCountAndContent(int recordCount) throws SQLException {
+    Statement st = connection.createStatement();
+    ResultSet rs = st.executeQuery("SELECT * FROM " + KAFKA_TOPIC);
+    int counter = 0;
+    while (rs.next()) {
+      Assert.assertEquals(counter++, rs.getInt("userId"));
+      Assert.assertEquals("Alex", rs.getString("firstName"));
+      Assert.assertEquals("Smith", rs.getString("lastName"));
+      Assert.assertEquals(20, rs.getInt("age"));
+    }
+    Assert.assertEquals(counter, recordCount);
+  }
+
+  private void sendTestDataToKafka(int startIndex, int numRecords) throws InterruptedException {
+    for (int i = startIndex; i < startIndex + numRecords; i++) {
+      String value = getTestKafkaRecord(KAFKA_TOPIC, SCHEMA, i);
+      connect.kafka().produce(KAFKA_TOPIC, null, value);
+    }
+  }
+
+  private Map<String, String> getConnectorConfigurations() {
```

Comment:
you can merge this with method `getSourceConnectorProps` and keep it in this class itself.
Similarly for `getSinkConnectorProps` in BaseIT.